pinot-common/src/main/java/com/linkedin/pinot/common/data/Schema.java

```diff
@@ -94,236 +89,241 @@ public static ZNRecord toZNRecord(Schema schema)
     return record;
   }
 
-  public Schema() {
-    dimensions = new ArrayList<String>();
-    metrics = new ArrayList<String>();
-    dimensionFieldSpecs = new ArrayList<DimensionFieldSpec>();
-    metricFieldSpecs = new ArrayList<MetricFieldSpec>();
-    timeFieldSpec = null;
-    fieldSpecMap = new HashMap<>();
+  public static String toSchemaString(Schema schema)
+      throws IOException {
+    String schemaString = schema.getJSONSchema();
+
+    // Throw out the exception if it happens.
+    if (schemaString != null) {
+      return schemaString;
+    } else {
+      return MAPPER.writeValueAsString(schema);
+    }
   }
 
-  public List<MetricFieldSpec> getMetricFieldSpecs() {
-    return metricFieldSpecs;
+  public String getSchemaName() {
+    return schemaName;
   }
 
-  public void setMetricFieldSpecs(List<MetricFieldSpec> metricFieldSpecs) {
-    for (MetricFieldSpec spec : metricFieldSpecs) {
-      addField(spec.getName(), spec);
-    }
+  public void setSchemaName(String schemaName) {
+    Preconditions.checkNotNull(schemaName);
+
+    this.schemaName = schemaName;
   }
 
   public List<DimensionFieldSpec> getDimensionFieldSpecs() {
     return dimensionFieldSpecs;
   }
 
   public void setDimensionFieldSpecs(List<DimensionFieldSpec> dimensionFieldSpecs) {
```

Comment:
rename methods to addAllDimensionFieldSpecs.

There is difference in set vs add. This method is doing add